website/static/js/osfToggleheight.js

```diff
@@ -0,0 +1,84 @@
+
+'use strict';
+var jQuery = require('jquery');
+
+(function ( $, window, document, undefined ) {
+
+	// Create the defaults once
+	var pluginName = 'osfToggleHeight';
+	var defaults = {
+			height:  50,
+			iconDown : 'fa fa-angle-down',
+			iconUp : 'fa fa-angle-up'
+	};
+
+	// The actual plugin constructor
+	function ToggleHeight ( element, options ) {
+		var self = this;
+		self.element = element;
+		self.$el = $(element);
+		self.settings = $.extend( {}, defaults, options );
+		self._defaults = defaults;
+		self._name = pluginName;
+
+		self.init =  function () {
+		    self.collapsed = true;
+			self.showToggle = false;
+			self.gradientDiv = $('<div class="' + self.element.id + '-gradient toggle-height-gradient" style="display:none"></div>').appendTo(self.element);
+			self.toggleDiv = $('<div class="' + self.element.id + '-toggle toggle-height-toggle text-center" style="display:none"></div>').insertAfter(self.element);
+		    $('.' + self.element.id + '-toggle.toggle-height-toggle').click(self.toggle);
+		    self.checkCollapse();
+		    $(window).resize(self.checkCollapse);
+		};
+		self.noToggle = function () {
+	        self.$el.css('height', 'auto');
+	        self.gradientDiv.hide();
+	        self.toggleDiv.hide();
+	        self.showToggle = false;
+	        self.$el.removeClass('toggle-height-parent');
+	    },
+	    self.toggleCollapse = function () {
```

Comment:
Minor, but these method names could use improvement. I think the following changes will bring the names close to the functionality:

- `noToggle` -> `hide` or `disable`
- `toggleCollapse` -> `collapse`
- `toggleOpen` -> `expand`
- `checkCollapse` -> `toggle`
- Remove `toggle` (see previous comment)?